src/triage/component/architect/label_generators.py

```diff
@@ -83,9 +108,35 @@ def generate_all_labels(
                 )
         nrows = [
             row[0] for row in
-            self.db_engine.execute('select count(*) from {}'.format(labels_table))
+            self.db_engine.execute(
+                'select count(*) from {}'.format(labels_table))
         ][0]
         if nrows == 0:
-            logging.warning('Done creating labels, but no rows in labels table!')
+            logging.warning(
+                'Done creating labels, but no rows in labels table!')
         else:
             logging.info('Rows in labels table: %s', nrows)
+
+
+class QueryBinaryLabelGenerator(BinaryLabelBase):
+    def generate(self, query, start_date, label_timespan, labels_table):
```

Comment:
The Experiment is going to need all the subclasses to have the same `generate` interface, so we'll have to take in the query at object construction time.  We'll also need to interpolate the start_date and label_timespan (I had forgotten about this one!) into the query.